app/src/app/components/sidebar/ShareMapsModal.tsx

```diff
@@ -0,0 +1,187 @@
+import {DocumentMetadata, DocumentObject} from '../../utils/api/apiHandlers';
+import {styled} from '@stitches/react';
+import {metadata} from '@/app/utils/api/mutations';
+type NamedDocumentObject = DocumentObject & {name?: string};
+
+const DialogContentContainer = styled(Dialog.Content);
+
+const BoxContainer = styled(Box, {
+  display: 'flex',
+  flexDirection: 'column',
+  gap: '1rem',
+});
+
+export const ShareMapsModal = () => {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+  const mapDocument = useMapStore(store => store.mapDocument);
+  const setMapDocument = useMapStore(store => store.setMapDocument);
+  const gerryDBTable = mapDocument?.gerrydb_table;
+  const userMaps = useMapStore(store => store.userMaps);
+  const updateMetadata = useMapStore(store => store.updateMetadata);
+  const [dialogOpen, setDialogOpen] = React.useState(false);
+  const [clickToCopyPrompt, setClickToCopyPrompt] = React.useState('Click to copy');
+
+  const handleClickToCopy = () => {
+    navigator.clipboard.writeText(
+      `${window.location.origin}${window.location.pathname}?document_id=${mapDocument?.document_id}`
+    );
+    setClickToCopyPrompt('Copied!');
+    setTimeout(() => {
+      setClickToCopyPrompt('Click to copy');
+    }, 2000);
+  };
+
+  const handleMapDocument = (data: NamedDocumentObject) => {
+    setMapDocument(data);
+    const urlParams = new URLSearchParams(searchParams.toString());
+    urlParams.set('document_id', data.document_id);
+    router.push(pathname + '?' + urlParams.toString());
+    // close dialog
+    setDialogOpen(false);
+  };
+
+  // get map name from metadata if it exists
+  const mapName = mapDocument?.metadata?.find(k => k.key === 'name')?.value || undefined;
+  const mapTags = mapDocument?.metadata?.find(k => k.key === 'tags')?.value || undefined;
+
+  const [name, setName] = React.useState(mapName);
+  const [tagsTeam, setTagsTeam] = React.useState(mapTags);
+  const [nameIsSaved, setNameIsSaved] = React.useState(false);
+  const [tagsIsSaved, setTagsIsSaved] = React.useState(false);
+
+  const handleChangeName = (name?: string) => {
+    if (mapDocument?.document_id) {
+      setName(name);
+      updateMetadata(mapDocument?.document_id, 'name', name);
+      // if name does not match metadata, make eligible to save
+      if (name !== mapName) {
+        setNameIsSaved(false);
+      }
+    }
+  };
+
+  const handleChangeTag = (tag?: string) => {
+    if (mapDocument?.document_id) {
+      setTagsTeam(tag);
+      updateMetadata(mapDocument?.document_id, 'tags', tag);
+
+      if (tag !== mapTags) {
+        setTagsIsSaved(false);
+      }
+    }
+  };
+
+  // if no gerrydb table selected or no edits made, return null
+  if (!gerryDBTable || !userMaps?.length) {
+    return null;
+  }
+
+  const handleMapSave = () => {
+    const metadataObjects = [
+      {key: 'name', value: name},
+      {key: 'tags', value: tagsTeam},
```

Comment:
H: Maybe add an explicit 'saved' property to help differentiate all recents vs saved ones `{key: 'saved', value: 'true'}` or similar